### PR TITLE
feat: purge cluster in between test methods; keep container runtime active throughout

### LINKED ISSUE #19182: [CPT] I can reuse the test environment
## Description

In Camunda Process Test, I want to have the option to reuse the test environment to speed up the test execution.

## Tasks
- Provide an option to use the test environment for multiple test executions 
- Possible levels of isolation:
    - No isolation between test executions
    - Isolate by using multi-tenancy 
    - Isolate by removing runtime and historical data  

requires #18997

### PR DESCRIPTION
**Pull request by Christian-Broomfield**
## Description

Currently, the CamundaProcessTest (Java) and CamundaSpringProcessTest (Spring) create and destroy a testcontainer runtime in between every test, inflating the runtime of these tests. With the implementation of the purge cluster operation we can keep the container runtime alive throughout the test lifecycle and only need to purge the cluster in between tests. 

closes #19182 

## Developer Notes

Spring technically has the `prepareTestInstance` override that lets us inject the methods immediately after `beforeTestClass` is called. This would be a nice optimization, but I opted not to do it because the implementation of the client creation would differ between the Java and Spring test extensions. Instead, both versions create new clients before each test and close them after each test. While it's only an edge case, this also prevents any shenanigans with users attempting to use a client to communicate with the cluster while it's (maybe) being purged. 

Also, the **CamundaProcessTestExtensionIT** fails to run locally on my machine because of the camunda/connectors-bundle container. The tests don't seem to indicate any problems with the integration tests, but I'm also unsure if they're even run.

### CODE REVIEW

**Review by saig0 — CHANGES_REQUESTED** [review]
@Christian-Broomfield the implementation looks good. :+1: 

But the test cases in `io.camunda.process.test.api.ExecutionListenerTest` are failing. And, I have some minor suggestions. Please have a look. :cookie:

File: testing/camunda-process-test-java/src/test/java/io/camunda/process/test/api/JunitExtensionTest.java [hunk 1]
```diff
@@ -207,8 +217,9 @@ void shouldStartAndCloseRuntime() throws Exception {
         new CamundaProcessTestExtension(camundaContainerRuntimeBuilder, NOOP);
 
     // when
+    extension.beforeAll(extensionContext);
```
[the remark this answers is not in the document]
Comment (Christian-Broomfield) [reply testing/camunda-process-test-java/src/test/java/io/camunda/process/test/api/JunitExtensionTest.java:220]: You're right! I've added a case in both the java and spring tests.

File: testing/camunda-process-test-spring/src/main/java/io/camunda/process/test/api/CamundaProcessTestExecutionListener.java [hunk 2]
```diff
@@ -57,8 +59,10 @@
  * <ul>
  *   <li>Publish a {@link CamundaClientClosingEvent}
  *   <li>Close created {@link CamundaClient}s
- *   <li>Stop the runtime
+ *   <li>Purge the Zeebe cluster
```
Comment (saig0) [inline testing/camunda-process-test-spring/src/main/java/io/camunda/process/test/api/CamundaProcessTestExecutionListener.java:62]: :wrench: Align JavaDoc with JUnit extension.

```suggestion
 *   <li>Purge the runtime (i.e. delete all data)
```

**Review by saig0 — APPROVED** [review]
@Christian-Broomfield looks good. :tada: 

One minor suggestion for the JavaDoc. Please have a look before merging. :rocket:
